File: dotnet/deployments/gcp-cloud-function/Function.cs
using core.controller;
using core.dao;
using core.mapper;
using gcp_cloud_function.mapper;
using Google.Cloud.Functions.Framework;
using Microsoft.AspNetCore.Http;

namespace gcp_cloud_function;

public class Function : IHttpFunction
{
    private static readonly GenericRequestMapper<HttpRequest> _requestMapper = new HttpRequestMapper(); 

    private static readonly PersonDao _personDao;

    private static readonly PersonController _personController;

    static Function()
    {
        _personDao = new FirestorePersonDaoImpl();
        _personController = new PersonController(_personDao);
    }
    public async Task HandleAsync(HttpContext context)
    {
        Console.WriteLine("Inside HandleAsync");

        try
        {
            var request = await _requestMapper.Map(context.Request);
            Console.WriteLine("{0}", request);
            
            Console.WriteLine("Calling personController#HandleRequest");
            var response = await _personController.HandleRequest(request);
            Console.WriteLine("{0}", response);
            
            // Map GenericResponse to context#Response.
            Console.WriteLine("Mapping GenericResponse to context object");
            context.Response.StatusCode = response.Status;
            Console.WriteLine("Successfully set status code");
            foreach (var keyValuePair in response.Headers)
            {
                context.Response.Headers.Append(keyValuePair.Key, keyValuePair.Value);
            }
            Console.WriteLine("Successfully set headers");
            await context.Response.WriteAsync(response.Body);
            Console.WriteLine("Successfully set body");
        }
        catch (Exception e)
        {
            Console.WriteLine(e);
            context.Response.StatusCode = 500;
            
            await context.Response.WriteAsync("");
        }
    }

}

File: java/deployments/azure-function/src/main/java/com/amazon/Function.java
package com.amazon;

import com.amazon.controller.PersonController;
import com.amazon.dao.CosmosDbPersonDaoImpl;
import com.amazon.dao.PersonDao;
import com.amazon.mapper.GenericResponse;
import com.amazon.mapper.HttpRequestMessageMapper;
import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.AuthorizationLevel;
import com.microsoft.azure.functions.annotation.FunctionName;
import com.microsoft.azure.functions.annotation.HttpTrigger;

import java.util.Optional;
import java.util.logging.Level;
import java.util.logging.LogRecord;

/**
 * Azure Functions with HTTP Trigger.
 */
public class Function {

    private static final HttpRequestMessageMapper httpRequestMessageMapper = new HttpRequestMessageMapper();

    private static final CosmosDbPersonDaoImpl personDao;

    private static final PersonController personController;

    static {
        personDao = new CosmosDbPersonDaoImpl();
        personController = new PersonController(personDao);
    }

    @FunctionName("personApiJava")
    public HttpResponseMessage run(
            @HttpTrigger(name = "req", methods = {HttpMethod.GET, HttpMethod.POST}, authLevel = AuthorizationLevel.ANONYMOUS, route = "person") HttpRequestMessage<Optional<String>> request,
            final ExecutionContext context
    ) {
        context.getLogger().info("Java HTTP trigger processed a request.");

        personDao.setLogger(context.getLogger());

        var genericResponse = new GenericResponse();
        try {
            context.getLogger().info("Mapping request");
            var genericRequest = httpRequestMessageMapper.map(request);

            context.getLogger().info(genericRequest.toString());
            context.getLogger().info("Calling controller");
            personController.handleRequest(genericRequest, genericResponse);

            context.getLogger().info("Building response");
            return this.buildHttpResponseMessage(
                    request.createResponseBuilder(HttpStatusType.custom(genericResponse.getStatus())),
                    genericResponse
            );


        }
        catch (Exception e) {
            context.getLogger().log(Level.SEVERE, e.getMessage());

            var httpStatusCode = genericResponse.getStatus() != 0 ? genericResponse.getStatus() : 500;

            return this.buildHttpResponseMessage(
                    request.createResponseBuilder(HttpStatusType.custom(httpStatusCode)),
                    genericResponse
            );
        }
    }

    private HttpResponseMessage buildHttpResponseMessage(HttpResponseMessage.Builder httpResponseMessageBuilder, GenericResponse genericResponse) {
        var responseBuilder = httpResponseMessageBuilder.body(genericResponse.getBody());

        genericResponse.getHeaders().forEach(responseBuilder::header);

        return responseBuilder.build();
    }
}
